lib/ansible/utils/path.py

```diff
@@ -35,3 +37,11 @@ def unfrackpath(path):
     '''
     return os.path.normpath(os.path.realpath(os.path.expandvars(os.path.expanduser(path))))
 
+def makedirs_safe(path, mode=None):
+    '''Safe way to create dirs in muliprocess/thread environments'''
+    if not os.path.exists(path):
```

Comment:
This introduces a race condition. Why check for existence here? What does it add?